vsphere/datadog_checks/vsphere/api.py

```diff
@@ -106,21 +125,38 @@ def smart_connect(self):
                 host=self.config.hostname, user=self.config.username, pwd=self.config.password, sslContext=context
             )
             # Next line tries a simple API call to check the health of the connection.
-            conn.CurrentTime()
+            version_info = VersionInfo(conn.content.about)
         except Exception as e:
             err_msg = "Connection to {} failed: {}".format(self.config.hostname, e)
             raise APIConnectionError(err_msg)
 
+        if not version_info.is_vcenter():
+            # Connection was successful but to something that is not a VirtualCenter instance. The check won't
+            # run correctly.
+            # TODO: Raise an exception and stop execution here
+            self.log.error(
```

Comment:
should it raise an exception?

Reply:
I'm leaning on the safe side and not raising the exception here. I can't be certain that all versions of vSphere that we are compatible with will give the type we are expecting. By raising an exception we might break existing users